python/pyspark/serializers.py

```diff
@@ -289,7 +291,6 @@ def _create_batch(self, series):
         def create_array(s, t):
             mask = s.isnull()
             # Ensure timestamp series are in expected form for Spark internal representation
-            # TODO: maybe don't need None check anymore as of Arrow 0.9.1
             if t is not None and pa.types.is_timestamp(t):
                 s = _check_series_convert_timestamps_internal(s.fillna(0), self._timezone)
                 # TODO: need cast after Arrow conversion, ns values cause error with pandas 0.19.2
```

Comment:
Do we still need the workaround `.from_pandas` .. `.cast` here?

Reply:
I didn't want to change this since it was related to a pandas version, I can double-check though